src/table.rs

```diff
@@ -33,6 +40,10 @@ pub enum Item {
     Table(Table),
     /// Type representing array of tables.
     ArrayOfTables(ArrayOfTables),
+    /// Type for display walker to know where (during walk) to
+    /// print dotted key. To get value, parse
+    /// the key and descend table tree.
+    DottedKeyMarker(Vec<SimpleKey>),
```

Comment:
This is public API, if we do `table['key'] = Item::DottedKeyMarker(vec!['a'.parse().unwrap(), 'b'.parse().unwrap()]);` 
what would it yield?
This part is the biggest concern with this PR.